### PR TITLE
Added type annotations to streams.py and message.py

### PR DESCRIPTION
**Pull request by HivaMohammadzadeh1**
I added the type annotations to Streams.py and message.py and tested it from examples/simple_pipeline.py by adding and running the following code:
def f() -> Stream[int]:
    pass
Still trying to figure out if I can add type annotations to split_by_type

### CODE REVIEW

**Comment by sukritkalra** [comment]
Hi! Thanks for the PR, could you please remove your environment from the commit so we could look at the changed files? Maybe try the following command: `git rm --cached -r my-env`

**Comment by sukritkalra** [comment]
No need for that, just add a new commit, and we can squash and merge the commits once the PR is ready.

**Comment by HivaMohammadzadeh1** [comment]
I just did. See if it's fixed. Sorry for the inconvenience.

File: python/erdos/message.py
Comment (sukritkalra) [inline python/erdos/message.py:62]:
```suggestion
class WatermarkMessage(Message[None]):
```
The `WatermarkMessage` is a special type of the `Message` class where the data being carried is empty, so we annotate that accordingly.

File: python/erdos/message.py
Comment (sukritkalra) [inline python/erdos/message.py:74]:
```suggestion
    def __str__(self):
```
The `__str__` method returns a `str` and not `T`. However, I don't think we should annotate dunder methods, just because their types are usually well known.

File: python/examples/simple_pipeline.py
Comment (sukritkalra) [inline python/examples/simple_pipeline.py:60]: This code (and the corresponding `Stream` import) are not being used in the example. Please remove.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:150]: Please remove.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:55]:
```suggestion
        self._internal_stream: PyStream = internal_stream
```
The `internal_stream` carries the same type that it is being assigned from.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:258]: Same comment as the above `internal_stream`.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:236]:
```suggestion
class ReadStream(Generic[T]):
```
I believe the scope of `TypeVar`s is restricted to the class, so the type variable `T` can be reused here.

Comment (sukritkalra) [reply python/erdos/streams.py:236]: Please also annotate the `Message` types in both `read` and `try_read` below.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:306]: This variable has the type `PyWriteStream` as discussed above.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:415]: Reuse the type variable `T`.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:430]: Please annotate `OperatorStream`.

Comment (sukritkalra) [reply python/erdos/streams.py:430]: Also annotate the type variable for `Message` in `read` below.

File: python/erdos/streams.py
Comment (sukritkalra) [reply python/erdos/streams.py:150]: The `split_by_type` function is a tricky one to provide types for. IMO I would go for the following:

```python
def split_by_type(self, *data_type: Type) -> Tuple["OperatorStream[Any]", ...]:
```

**Review by sukritkalra — CHANGES_REQUESTED** [review]
Thanks for the PR. Please take a look at my comments:

File: python/examples/simple_pipeline.py
Comment (HivaMohammadzadeh1) [reply python/examples/simple_pipeline.py:60]: I added it to check my annotations. Sure, will do.

File: python/erdos/streams.py
Comment (HivaMohammadzadeh1) [reply python/erdos/streams.py:150]: Not sure about my solution , but gave it a try.

**Comment by HivaMohammadzadeh1** [comment]
Thank you for your comments. I have considered all the comments and changed my code accordingly. Please let me know if there's anything that I still have to fix. 
Also, for the function "split_by_type", I attempted to add type annotations to the code, but I'm not sure if it is fixed.

File: python/erdos/message.py
Comment (sukritkalra) [inline python/erdos/message.py:2]:
```suggestion
from typing import Generic, TypeVar
```

`Any` is not being used anymore. Please remove.

File: python/erdos/message.py
Comment (sukritkalra) [inline python/erdos/message.py:74]: I'm not sure why the CI is not running. But, could you please run the following from your `python` directory?

```bash
black .
isort --profile black .
flake8 --inline-quotes=double ./
```
and commit any changes that these tools make?

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:146]:
```suggestion
    def split_by_type(self, *data_type: Type) -> Tuple["OperatorStream[Any]", ...]:
```

File: python/examples/simple_pipeline.py
Comment (sukritkalra) [inline python/examples/simple_pipeline.py:13]:
```suggestion
from erdos.streams import ReadStream, WriteStream
```

**Review by sukritkalra — CHANGES_REQUESTED** [review]
Thanks for the changes! Please run the tools as I've suggested, and we can wait for @pschafhalter to approve the changes too.

File: python/erdos/message.py
Comment (pschafhalter) [reply python/erdos/message.py:74]: We need to manually approve and run the CI because Hiva is a first-time committer.

Comment (sukritkalra) [reply python/erdos/message.py:74]: Thanks! @HivaMohammadzadeh1 the CI should now show the errors in formatting, please take a look and address the issues.

Comment (HivaMohammadzadeh1) [reply python/erdos/message.py:74]: Thank you! Will do.

Comment (HivaMohammadzadeh1) [reply python/erdos/message.py:74]: After running `flake8 --inline-quotes=double ./`
I get this error. Is this normal?
```
usage: flake8 [options] file file ...
flake8: error: unrecognized arguments: --inline-quotes=double
```
It does not show me any errors in formatting, so I think that error might not be normal.

Comment (HivaMohammadzadeh1) [reply python/erdos/message.py]: Perfect that was the problem. I ran and resolved the issues and committed.

File: python/erdos/streams.py
Comment (pschafhalter) [inline python/erdos/streams.py:321]:
```suggestion
    def send(self, msg: Message[T]) -> None:
```

File: python/erdos/streams.py
Comment (pschafhalter) [inline python/erdos/streams.py:365]:
```suggestion
    def connect_loop(self, stream: OperatorStream[T]) -> None:
```

File: python/erdos/streams.py
Comment (pschafhalter) [inline python/erdos/streams.py:426]:
```suggestion
    def __init__(self, stream: OperatorStream[T]) -> None:
```

**Review by pschafhalter — CHANGES_REQUESTED** [review]
A few small nits. Happy to merge once they're addressed and the formatting checks pass!

**Comment by HivaMohammadzadeh1** [comment]
I just did the changes. The commands run on my end. I wonder why they don't on here. Thank you for being patient with me and helping me

**Comment by sukritkalra** [comment]
Hi! I had to manually approve the workflow. It seems like your formatter made some changes in `python/examples/linq.py` and `python/erdos/context.py` that are not correct. Could you run `black .` again locally, fix the files and commit the changes please?

**Comment by HivaMohammadzadeh1** [comment]
Just committed again :)

**Comment by HivaMohammadzadeh1** [comment]
Yes, I figured it was doing that, so I didn't run `black .` as that was what was formatting it.

**Comment by sukritkalra** [comment]
Looks like its still failing, are these changes being made by `black`? Are you running a special `black` profile?

**Comment by HivaMohammadzadeh1** [comment]
I just manually went and fixed it. Can you check if it works now? I didn't run `black .`

**Comment by sukritkalra** [comment]
Nope, still not working. I wonder why your `black` invocation is not fixing this. If you type `black .` on your command line, it should automatically reformat the `context.py` to what we had before.

**Comment by HivaMohammadzadeh1** [comment]
I just ran `black .` and committed to see if it's fixed

**Comment by sukritkalra** [comment]
Nope, just for reference, my black version is 22.6.0 and it seems to fix the errors being raised by the CI.

**Comment by HivaMohammadzadeh1** [comment]
I don't know why it's reformatting it wrong.

File: python/erdos/streams.py
Comment (sukritkalra) [inline python/erdos/streams.py:411]: I believe `ExtractStream` is also of type `Stream[T]`.

**Review by sukritkalra — CHANGES_REQUESTED** [review]
Fixed the formatting issues for you, but there's issues in the pythondoc that need to be fixed.

File: python/erdos/streams.py
Comment (HivaMohammadzadeh1) [reply python/erdos/streams.py:411]: Great thanks will fix this and push shortly

**Review by sukritkalra — APPROVED** [review]
LGTM. Had to add a nitpick to Sphinx for doc issues.

**Pull request merged**
